### ISSUE TITLE
Shadow appears on Simulator but not on Device

### ISSUE THREAD

**Issue by oisinlavery** [facebook/react-native#449]
iPhone 6, iOS 8.1.1

```javascript
var TestKit = React.createClass({
  render: function() {
    return (
      <View style={styles.main}>
        <Box/>
      </View>
    );
  }
});

var Box = React.createClass({
  render: function() {
    return (
      <View style={styles.box}></View>
    );
  }
});

var styles = StyleSheet.create({
  main: {
    flex: 1,
    backgroundColor: "#f1f1f1",
    justifyContent: "center",
    alignItems: "center"
  },
  box: {
    height: 100,
    width: 100,
    backgroundColor: "#ffffff",
    borderRadius: 10,
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 0
    }
  }
});
```
![0](https://cloud.githubusercontent.com/assets/1757344/6886837/8767980c-d600-11e4-976c-202483c8fe9c.jpg)

**Comment by vjeux** [comment]
cc @sahrens If I remember correctly, it's not rendering shadows for phones that are not good enough and by default the simulator fits that.

**Comment by NovaDev94** [comment]
Some images for this, the shadows is rendered fine on 4s, 5 but not 5s and 6:
![screen shot 2015-03-30 at 1 40 33 am](https://cloud.githubusercontent.com/assets/6073918/6887079/01fd8acc-d67e-11e4-8d52-659be87cc4fe.png)
![screen shot 2015-03-30 at 1 41 00 am](https://cloud.githubusercontent.com/assets/6073918/6887081/025c1a1a-d67e-11e4-8884-022fe2aa5a4e.png)
![screen shot 2015-03-30 at 1 40 48 am](https://cloud.githubusercontent.com/assets/6073918/6887080/022dd074-d67e-11e4-964a-cb690673dc7e.png)
![screen shot 2015-03-30 at 1 41 10 am](https://cloud.githubusercontent.com/assets/6073918/6887082/028d740c-d67e-11e4-9e9f-886cd5975e80.png)

**Comment by oisinlavery** [comment]
Shadow is NOT visible on iPhone 6

Shadow is visible on:
- iPhone 5 Simulator
- iPad Retina Simulator
- iPad 2 Simulator

Shadow is NOT visible on
- iPhone 5s Simulator
- iPhone 6 Simulator
- iPhone 6+ Simulator
- iPad Air Simulator

**Comment by sahrens** [comment]
Our GPU fill-rate gating was only for some specific product code, and was no longer necessary after we switched to assets for shadows instead of the layer properties.  There shouldn't be anything in the infra intentionally disabling shadows, probably just some sort of bug - cc @a2, @nicklockwood

That said, I would recommend using assets instead of the shadow properties in general - they tend to perform much better on older devices.

**Comment by NovaDev94** [comment]
The problem is the shadow properties only work on old devices, not the new
ones.

On Monday, March 30, 2015, Spencer Ahrens <notifications@github.com> wrote:

> Our GPU fill-rate gating was only for some specific product code, and was
> no longer necessary after we switched to assets for shadows instead of the
> layer properties. There shouldn't be anything in the infra intentionally
> disabling shadows, probably just some sort of bug - cc @a2, @nicklockwood
>
> That said, I would recommend using assets instead of the shadow properties
> in general - they tend to perform much better on older devices.
>
>
> > On Mar 29, 2015, at 12:27 PM, Oisin Lavery <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> >
> > Shadow is NOT visible on iPhone 6
> >
> > Shadow is visible on:
> >
> > iPhone 5 Simulator
> > iPad Retina Simulator
> > iPad 2 Simulator
> > Shadow is NOT visible on
> >
> > iPhone 5s Simulator
> > iPhone 6 Simulator
> > iPhone 6+ Simulator
> > iPad Air Simulator
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/react-native/issues/449#issuecomment-87484660>
> .
>


-- 
Regards,
Truong Ngoc Dai

**Comment by brentvatne** [comment]
:+1: I noticed this today and would be happy to attempt to fix the bug if someone could point me in the right direction.

**Comment by wangii** [comment]
as far as I get, `setShadowOpacity` mysteriously not working inside of `RCTSetProperty` on iPhone simulator running iPhone 5s, iPhone 6, and iPhone 6 plus.

**Comment by mbrgm** [comment]
What's the current state of this?

**Comment by jsierles** [comment]
I see the shadow now on the iPhone 5s simulator: https://rnplay.org/plays/6_nkSA

Also see the shadow on the iPhone 6 device.

@oisinlavery Still seeing this issue?

**Comment by brentvatne** [comment]
I see the shadow on my iPhone 5s. Let's reopen this if anyone has the issue again.

**Comment by mbrgm** [comment]
I found that the shadow disappears once I set `overflow: 'hidden'` in the view's style. Is this intended behavior?

**Comment by sahrens** [comment]
The shadow typically displays outside the bounds of the view, so if you set
overflow hidden I would expect it to clip the shadow. If you really need
overflow hidden, you could put the shadow properties on a wrapper view that
doesn't hide the overflow?

**Comment by mbrgm** [comment]
I expected it to clip the overflowing content of the view (like when you have rounded corners), but not the shadow. Apparently this is a very subjective expectation and having a container view for the shadow properties works just as well.

**Comment by vjeux** [comment]
@mbrgm: our base reference to all of those questions is the web. Can you add a shadow and overflow hidden on a div in a jsfiddle and see what happens?

**Comment by mbrgm** [comment]
@vjeux As I expected, the shadow is visible.

https://jsfiddle.net/L55drqzg/1/

**Comment by vjeux** [comment]
cc @nicklockwood

**Comment by sahrens** [comment]
Yeah, we should try to follow web where possible, but this might be kinda tricky - what do you think, @nicklockwood?

**Comment by nicklockwood** [comment]
@sahrens I agree, we should try to emulate CSS, as we do for all other aspects of border rendering.

There's a fix I need to make to clipping that will fix this as a side effect anyway.

The best solution in the meantime is to nest two views, with the shadow on the outer view and clipping on the inner view (this is basically what my fix will be, but it'll be done opaquely by the framework instead of explicitly).

**Comment by aleclarson** [comment]
What's the status on this?

**Comment by nicklockwood** [comment]
I'm looking into addressing this, but it will take a while. FWIW, It's not a bug; this is how shadows are supposed to work on iOS, and there are fairly simple workarounds (nesting two views). But we would ideally like React to conform to the CSS spec here, so I'm working to make that happen.

**Comment by nicklockwood** [comment]
Sorry, I see I already said allthat earlier ¯\_(ツ)_/¯

**Comment by eyaleizenberg** [comment]
Any updates on this? Thanks...

**Comment by Amurmurmur** [comment]
Any updates on that? Still experiencing it pon iOS 10.3.3, iPhone 7, overflow: 'hidden' no shadows :/

**Comment by HAMPLENI** [comment]
The workaround to have the shadow style properties in the container `View` and the `overflow: "hidden"` style in the inner view is not working for me, perhaps because my container View needs to have `position: "absolute"` on it.

What I see visually is that the `overflow: "hidden"` style is ignored and I have the proper shadows.

**Comment by HAMPLENI** [comment]
If anyone runs into this when dealing with an `Animated.View`, make sure you make both `View` components be an `Animated.View` component.  You will also need to set the height on the contained `Animated.View` in order for `overflow: hidden` to work on it.

**Comment by deb0ch** [comment]
Still happening as of today 😕

**Comment by nanndoj** [comment]
+1
Still happening today no shadows when using `overflow: hidden`
